src/program/operations/save/index.js

```diff
@@ -63,36 +106,63 @@ const run = args => {
       });
     } else {
       // Save script
-      saveYamlFile(yamlFileName, yamlFile)
-        .then(() => {
-          GlobalConfig.addScriptName(scriptName);
-          GlobalConfig.save();
-          printMessage(formatMessage(messages.savedScript, { scriptName }));
-        })
+      saveYamlFile(fileName, yamlFile)
+        .then(() => saveScript(scriptName))
+        .catch(error => {
+          printMessage(formatMessage(messages.errorSavingScript, { scriptName, error }));
+        });
+    }
+  } else if (isValidJsonFileName(fileName)) {
+    const jsonFile = loadJsonFile(fileName);
+    const scriptName = getFirstKey(jsonFile);
+
+    GlobalConfig.load();
+    if (GlobalConfig.hasScript(scriptName)) {
+      shouldUpdateScript(scriptName).then(answer => {
+        if (answer) {
+          // Delete previous script
```

Comment:
nit:
- [ ] `Delete previous script` => `Delete any previously saved scripts with the same name`